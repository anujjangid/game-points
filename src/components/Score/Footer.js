import React, { useContext } from 'react';

import { getBonus, getTotal } from '../../utils';
import { BonusText, TotalText } from '../../Constants';
import { CounterContext } from '../../reducer';
import { ScoreFooter, ScoreItem } from './styles';

const Footer = props => {
  const { state } = useContext(CounterContext);

  return (
    <ScoreFooter>
      <ScoreItem>
        <span>{BonusText}</span>
        <span>{(state && state.items && getBonus(state.items)) || 0}</span>
      </ScoreItem>
      <ScoreItem>
        <span>{TotalText}</span>
        <span>{(state && state.items && getTotal(state.items)) || 0}</span>
      </ScoreItem>
    </ScoreFooter>
  );
};

export default Footer;
