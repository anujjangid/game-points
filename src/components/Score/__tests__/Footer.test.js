import React from 'react';
import { create } from 'react-test-renderer';
import Footer from '../Footer';

const props = {
  data: { price: 10, game: 'A', count: 1 }
};
describe('Test suite for Score Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions

  it('render correctly', () => {
    const component = create(<Footer {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
    const rootInstance = component.root;
    // check heading
    const FooterItems = rootInstance.findByType('footer');
    expect(FooterItems.props.children).toHaveLength(2);
  });
});
