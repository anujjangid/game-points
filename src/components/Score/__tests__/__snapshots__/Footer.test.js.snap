// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Score Component render correctly 1`] = `
<footer
  className="sc-htpNat nCSaE"
>
  <div
    className="sc-bxivhb iMiCKC"
  >
    <span>
      Bonus
    </span>
    <span>
      0
    </span>
  </div>
  <div
    className="sc-bxivhb iMiCKC"
  >
    <span>
      Total
    </span>
    <span>
      0
    </span>
  </div>
</footer>
`;
